### DISCUSSION TITLE
如何优化TreeGraph的卡顿问题？

### DISCUSSION THREAD

**Discussion by zhangbaojia** [antvis/G6#5657] [Q&A] [answered]
新建了干净的vue项目只用来测试G6性能。

使用treeGraph测试发现：
1、层级：4        节点数：5        效果：不卡
2、层级：5        节点数：5        效果：卡顿
这个如何优化？

示例代码:
<img width="391" alt="c95e9cbe624e280a321e56f72693feb" src="https://github.com/antvis/G6/assets/19427334/3feda8b2-fa43-40ec-9bdd-d88536d053ae">
<img width="413" alt="994f8c31f88cb443b9e162657fb94c2" src="https://github.com/antvis/G6/assets/19427334/efef793e-6066-4fae-8078-8e0926bcf563">

**Comment by hustcc** [comment]
能给个预览地址吗？@zhangbaojia

**Reply by zhangbaojia** [reply]
可以将app.vue文件中 - line: 112行里面的参数： 4 改成 5 就开始卡顿了。
(界面内容是有的，数据稍大点，渲染的小，需要通过鼠标滚轮放大)
https://stackblitz.com/edit/vitejs-vite-g23sme?file=src%2FApp.vue

**Reply by zhangbaojia** [reply, marked as answer]
const graph = ref(null);
graph.value = new G6.TreeGraph({});  

是因为将graph丢给vue的ref变量管理导致的，先换成shallowRef试试
